src/resmom/job_recov_fs.c

```diff
@@ -195,8 +195,7 @@ job_save_fs(job *pjob)
 
 	} else {
 		/* an attribute changed,  update mtime */
-		pjob->ji_wattr[JOB_ATR_mtime].at_val.at_long = time_now;
-		pjob->ji_wattr[JOB_ATR_mtime].at_flags |= ATR_MOD_MCACHE;
+		set_attr_l(&pjob->ji_wattr[JOB_ATR_mtime], time_now, SET);
```

Comment:
@agrawalravi90 I haven't gone through all changes but can we make these setter like `set_jattr(pjob, ATTR_NAME, ...)` and internally it can access `&pjob->ji_wattr[ATTR_NAME]` now, and later if we ever change ji_wattr, from array to pointer (or something else), we just have to change at one place only

Also it will more inline with `get_jattr_*`

Reply:
I did add one called set_jattr_long, and I think it can be used here, so I'll convert this to set_jattr_long, thanks

Reply:
huh, i must have been dreaming, there isn't a set_jattr_long :) I can add that, thanks.

Reply:
Well my point was not only for this one, but in general for all job attr setters...
The reason I am asking is we have plan to change attributes array on job struct (although not finalized) to reduce mem footprint of svr, so if we can make these setters not to take attribute as argument but instead job object then it will be easy to change in future